### ISSUE TITLE
Refactor neonKUBE versioning

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#937]
We need to make some network related cluster definition changes:

- [x] Rename `IngressOptions --> IngressRoute`

     This is better because it better explains what this actually is.

- [x] Rename `NetworkOptions.Ingress --> IngressRoutes`

- [x] Remove the `NetworkOptions` properties below.  We're not going to allow the user to customize any of this.  The components and versions installed will be essentially baked into each neonKUBE release.

     `Cni`
     `CniVersion`
     `IstioVersion`

- [x] Replace `KubeClient` version compatibility stuff with fixed constants in a new static `KubeComponents` class.  We're baking in the sub-component releases into the each neonKUBE release so we no longer need to maintain this crapola.

- [x] Relocate the Kubernetes dashboard helm YAML from `HeadendClient` to where it's actually used.

- [ ] **JEFF:** Relocate the _resources_ from **neon-cli** to **Neon.Kube** as well as all of the cluster provisioning, preparation and setup related code.  The idea is to put this all in the library so we'll be able to use it from other places (like a cloud installation service).

- [x] **BREAKING** **JEFF:** Rename **Neon.Kube.Service** to **Neon.Service** and rename `KubeService --> NeonService` and `KubeServiceStatus --> NeonServiceStatus`.

- [ ] **JEFF:** We need to update the builds for images that will be deployed to clusters will be tagged with the neonKUBE cluster version.  We should probably do this for all foundational images like elasticsearch and kibana so it'll be super easy to know which images go with each neonKUBE release.

- [x] **JEFF:** Remove all **Neon.Kube.*** packages from the nuget publication scripts.

- [x] **JEFF:** Create the new **neonCLOUD** repo.

- [x] **JEFF:** Split `$/product-version.txt` into `neonKUBE-version.txt` and `neonLIBRARY-version.txt` do we can version these components differently and then update the build tools and scripts to use these files.

- [x] **JEFF:** Rename the neonKUBE releases to neonDESKTOP.

- [x] **JEFF:** Update the release instructions and templates to explain how to do the new releases.

- [x] **JEFF:** Setup builder scripts use the `$/kube-version.txt` file to determine which versions of Kubernetes client tools to download.  This is a bit confusing so we're going to rename this `kubernetes-version.txt`.

- [x] **JEFF:** Rename the **neonkube-cleaner** service to **neon-cleaner** and do the same for **neonkube-disable-thp**.  This will be consistent with **neon-bode-prep.service**.

- [x] **JEFF:** Investigate creating a **neonLIBRARY** repo and using that for nuget package release notes so we can move these notes out of neonKUBE and dedicate the release notes there for actual neonKUBE cluster releases.  The big question is whether the releases in **neonLIBRARY** will be able to reference issues in **neonKUBE**.  You know, this probable isn't going to work because we'd want the tags to reference the neonKUBE repo.  We'll probably need to start adding **neonLIBRARY** and **neonKUBE** to the release titles.

     **JEFF UPDATE:** This won't work.

- [ ] **MARCUS:** Refactor how Helm charts work.  We need to replace `HeadendClient.GetHelmChartZipAsync()` and `AddGitFilesToZipAsync()` with static text embedded in `Neon.Kube`.  Perhaps we can use the `resources` scheme we used for **neon-cli** scripts.

     **FYI:** The zipping you're doing in the old code is unnecessary.  `SshProxy<T>` supports the concept of a `CommandBundle`.  This is the combination of a script that will be run with any number of text or binary files.  The idea is that you create a bundle with the script you want to run and then you add any files to the bundle and then run the bundle normally or as SUDO.  Under the covers, `SshProxy<T>`:

     a. creates a zip archive including the script (which is implicitly named `__run.sh` along with any added files)
     b. uploads the zipped bundle to a secure temporary directory
     c. unzips the files into the directory
     d. changes the current directory to to the temporary one and then runs `__runs.h` (which can access the unzipped data files in the current directory)
     e. deletes the directory
     f. returns the exit code and stdout and stderr results

     This is much cleaner because it handles all of the zip stuff for you and it'll also be more secure because I ensure the temporary directory has permissions only for the **sysadmin** user.  We also have a cleaner service running on all nodes that removes temporary directories older than a day or two to ensure that node don't accumulate a bunch of old run scripts, potentially including sensitive information.  `CommandBundle` also handles the conversion of Windows to Linux line endings.  Here's an example:
```
     // Enable the public key by appending it to [$HOME/.ssh/authorized_keys],
     // creating the file if necessary.  Note that we're allowing only a single
     // authorized key.

     var addKeyScript =
$@"
chmod go-w ~/
mkdir -p $HOME/.ssh
chmod 700 $HOME/.ssh
touch $HOME/.ssh/authorized_keys
cat ssh-key.pub > $HOME/.ssh/authorized_keys
chmod 600 $HOME/.ssh/authorized_keys
";
     bundle = new CommandBundle("./addkeys.sh");

     bundle.AddFile("addkeys.sh", addKeyScript, isExecutable: true);
     bundle.AddFile("ssh-key.pub", kubeContextExtension.SshClientKey.PublicPUB);

     node.RunCommand(bundle);
```
     This example specifies the script file name explicitly.  `CommandBundle` includes some static methods that simplify specifying a script using the default name to save a line or two of code.

**Comment by jefflill** [comment]
...added @johncburns1 so he can follow along.

**Comment by jefflill** [comment]
Closing because this will be addressed in #1110
